src/rpc/blockchain.cpp

```diff
@@ -1558,11 +1607,53 @@ RPCHelpMan getblockchaininfo()
     SoftForkDescPushBack(tip, softforks, consensusParams, Consensus::DEPLOYMENT_SEGWIT);
     SoftForkDescPushBack(tip, softforks, consensusParams, Consensus::DEPLOYMENT_TESTDUMMY);
     SoftForkDescPushBack(tip, softforks, consensusParams, Consensus::DEPLOYMENT_TAPROOT);
-    obj.pushKV("softforks", softforks);
+    return softforks;
+}
+} // anon namespace
 
-    obj.pushKV("warnings", GetWarnings(false).original);
-    return obj;
-},
+static RPCHelpMan getdeploymentinfo()
+{
+    return RPCHelpMan{"getdeploymentinfo",
+        "Returns an object containing various state info regarding soft-forks.",
+        {
+            {"blockhash", RPCArg::Type::STR_HEX, RPCArg::Default{"chain tip"}, "The block hash at which to query fork state"},
```

Comment:
7f15c1841b98de6931a7ac68e16635a05d3e96cf current naming style for new rpc arguments, and a default description that looks a bit less like "chain tip" could be a value
```suggestion
            {"block_hash", RPCArg::Type::STR_HEX, RPCArg::Default{"value of chain tip"}, "The block hash at which to query fork state"},
```

It would be good to have tests for invoking getdeploymentinfo with a valid and an invalid block hash.

Reply:
This code is reused in the deprecated `getblockchaininfo` part, so let's not rename these things.

Reply:
Oh that's true; the functionality is nicely reused.

Reply:
Ugh, why are we introducing new naming styles that are incompatible with the existing names, especially where they're part of an API? :vomiting_face: